### PR TITLE
Add BCQuery unit test

### PR DESCRIPTION
**Pull request by xuanzhui**

### CODE REVIEW

**Comment by xuanzhui** [comment]
Class cn.beecloud.BCQueryTest

all > cn.beecloud > BCQueryTest
26
tests

0
failures

0
ignored

0.518s
duration

100%
successful

Tests
Tests

Test	Duration	Result
testQueryBillByIDAsync1	0s	passed
testQueryBillByIDAsync2	0.016s	passed
testQueryBillByIDAsync3	0.015s	passed
testQueryBillsAsync1	0.016s	passed
testQueryBillsAsync2	0.015s	passed
testQueryBillsAsync3	0.016s	passed
testQueryBillsAsync4	0s	passed
testQueryBillsAsync5	0.016s	passed
testQueryBillsAsync6	0.015s	passed
testQueryOfflineBillStatusAsync1	0s	passed
testQueryOfflineBillStatusAsync2	0.015s	passed
testQueryOfflineBillStatusAsync3	0.010s	passed
testQueryOfflineBillStatusAsync4	0.016s	passed
testQueryRefundByIDAsync1	0s	passed
testQueryRefundByIDAsync2	0.247s	passed
testQueryRefundByIDAsync3	0.037s	passed
testQueryRefundStatusAsync1	0s	passed
testQueryRefundStatusAsync2	0.016s	passed
testQueryRefundStatusAsync3	0s	passed
testQueryRefundStatusAsync4	0.016s	passed
testQueryRefundsAsync1	0s	passed
testQueryRefundsAsync2	0.019s	passed
testQueryRefundsAsync3	0.013s	passed
testQueryRefundsAsync4	0.005s	passed
testQueryRefundsAsync5	0.015s	passed
testQueryRefundsAsync6	0s	passed
Generated by Gradle 2.2.1 at 2015-11-17 10:01:14

**Comment by xuanzhui** [comment]
Class cn.beecloud.BCOfflinePayTest
all > cn.beecloud > BCOfflinePayTest
16
tests
0
failures
0
ignored
0.702s
duration
100%
successful
Tests
Test	Duration	Result
testReqAliOfflineQRCodeAsync	0.013s	passed
testReqOfflinePayAsyncByBean	0.012s	passed
testReqOfflinePayAsyncErrorFromServer	0.006s	passed
testReqOfflinePayAsyncNetworkError	0.009s	passed
testReqOfflinePayAsyncParamInvalid	0.009s	passed
testReqOfflinePayAsyncSucc	0.021s	passed
testReqQRCodeAsyncByBean	0.010s	passed
testReqQRCodeAsyncChannelInvalid	0s	passed
testReqQRCodeAsyncErrorFromServer	0.015s	passed
testReqQRCodeAsyncNetworkError	0.016s	passed
testReqQRCodeAsyncParamInvalid	0.006s	passed
testReqQRCodeAsyncSucc	0.028s	passed
testReqRevertBillAsyncNetworkError	0.005s	passed
testReqRevertBillAsyncParamInvalid	0.006s	passed
testReqRevertBillAsyncSucc	0.024s	passed
testReqWXNativeQRCodeAsync	0.522s	passed
Generated by Gradle 2.2.1 at 2015-11-17 20:45:27

File: sdk/src/main/java/cn/beecloud/entity/BCReqParams.java [hunk 1]
```diff
@@ -41,7 +41,7 @@
     /**
      * 请求类型
      */
-    public enum ReqType{PAY, QUERY, QRCODE, OFFLINE_PAY}
+    public enum ReqType{PAY, QUERY, QRCODE, OFFLINE, OFFLINE_PAY}
```
Comment (quietbamboo) [inline sdk/src/main/java/cn/beecloud/entity/BCReqParams.java]: 需要解释一下每一个ReqType分别什么含义。

File: sdk/src/main/java/cn/beecloud/entity/BCReqParams.java [hunk 2]
```diff
@@ -258,6 +258,17 @@ public static boolean isValidAPPPaymentChannelType(BCChannelTypes channel) {
          * @return true表示有效
          */
         public static boolean isValidOfflinePayChannelType(BCChannelTypes channel) {
+            return channel == WX_SCAN ||
+                    channel == ALI_SCAN;
```
Comment (quietbamboo) [inline sdk/src/main/java/cn/beecloud/entity/BCReqParams.java]: 这里写在一行，写成

return ((channel == WX_SCAN) || (channel == ALI_SCAN));

File: sdk/src/test/java/cn/beecloud/BCOfflinePayTest.java [hunk 3]
```diff
@@ -0,0 +1,725 @@
...
+
+                        Assert.assertTrue(revertStatus.getRevertStatus());
+
+                        latch.countDown();
+                    }
+                });
+
+        latch.await(2000, TimeUnit.MILLISECONDS);
+    }
+}
```
Comment (quietbamboo) [inline sdk/src/test/java/cn/beecloud/BCOfflinePayTest.java]: 文件结尾要有一个空行

**Comment by quietbamboo** [comment]
看了一遍，没有什么大问题。这个mock的方法挺好的。

**Pull request merged**